runner_test.go

```diff
@@ -116,6 +119,75 @@ func TestSpawnWorkers(t *testing.T) {
 	}
 }
 
+func TestSpawnWorkersWithManyTasks(t *testing.T) {
+	var lock sync.Mutex
+	taskCalls := map[string]int{}
+
+	createTask := func(name string, weight int) *Task {
+		return &Task{
+			Name:   name,
+			Weight: weight,
+			Fn: func() {
+				lock.Lock()
+				defer lock.Unlock()
+				taskCalls[name] += 1
+			},
+		}
+	}
+	tasks := []*Task{
+		createTask(`one hundred`, 100),
+		createTask(`ten`, 10),
+		createTask(`one`, 1),
+	}
+
+	runner := newSlaveRunner("localhost", 5557, tasks, nil)
+	defer runner.close()
+
+	runner.client = newClient("localhost", 5557, runner.nodeID)
+
+	const numToSpawn int = 30
+	const hatchRate float64 = 10
+	runner.hatchRate = hatchRate
+
+	go runner.spawnWorkers(numToSpawn, runner.stopChan, runner.hatchComplete)
+	time.Sleep(4 * time.Second)
+
+	currentClients := atomic.LoadInt32(&runner.numClients)
+
+	assert.Equal(t, numToSpawn, int(currentClients))
+	lock.Lock()
+	hundreds := taskCalls[`one hundred`]
+	tens := taskCalls[`ten`]
+	ones := taskCalls[`one`]
+	lock.Unlock()
+
+	total := hundreds + tens + ones
+	t.Logf("total tasks run: %d\n", total)
+
+	assert.True(t, total > 111)
+
+	assert.True(t, ones > 1)
+	actPercentage := float64(ones) / float64(total)
+	expectedPercentage := 1.0 / 111.0
```

Comment:
tl;dr this test is checking that even a task with weight `1` will get run in accordance to its weight